File: internal/analyze/kotlin/testdata/cdd_examples.kt
// docs/cdd.md section 2, verbatim rules
fun check(a: Int, b: Int, c: Int, d: Int): Boolean {
    if (a > b && c < d) {     // code_branch 1, condition 2
        return true
    }
    return false
}                             // total 3

fun guarded() {
    try { a() } catch (e: E) { b() } finally { c() }   // exception_handling 3
}                             // total 3

fun ifElse(x: Int) = if (x > 0) 1 else 2               // code_branch 2 (if-else = 2)

File: internal/analyze/golang/testdata/cdd_examples.go
// cdd_examples.go — docs/cdd.md section 2, in Go

package app

type Examples struct{}

func (Examples) Check(a, b, c, d int) bool {
	if a > b && c < d { // code_branch 1, condition 2
		return true
	}
	return false
}

func (Examples) IfElse(x int) int {
	if x > 0 { // code_branch 1
		return 1
	} else { // code_branch 1 — the alternative is a block
		return 2
	}
}

// unit Examples: struct, code_branch 3, condition 2, local_variable 0
